### DISCUSSION TITLE
Backend API

### DISCUSSION THREAD

**Discussion by efkz** [n1crack/vuefinder#53] [Q&A]
How to pair it with my custom server?

**Comment by n1crack** [comment]
A backend library is needed to use this code. You should check the [Vuefinder php library](https://github.com/n1crack/vuefinder-php) .

Unfortunately, there is no other backend library other than php right now. But I don't think its difficult to implement one.
If you don't use php in your server, you can make your own library with any language, with just looking at the php library (It is a single class: https://github.com/n1crack/vuefinder-php/blob/master/src/VueFinder.php)

Here is the live demo setup: 
https://github.com/n1crack/vuefinder.ozdemir.be/blob/fc341f39433b87ad81532ce22edf5862d728f0ef/routes/web.php#L1-L28

**Reply by efkz** [reply]
Do you have any fully functional example? Can't make it work ((

**Comment by n1crack** [comment]
The source code of vuefinder.ozdemir.be is here :  
https://github.com/n1crack/vuefinder.ozdemir.be

You can install it via git clone : 

```bash
git clone https://github.com/n1crack/vuefinder.ozdemir.be.git 
cd vuefinder.ozdemir.be
```

then you can install it with `composer install`

If you don't use php or laravel, then, unfortunately, there is no example at the moment.

**Comment by carueda** [comment]
@n1crack - your component looks very elegant and well-done! However, I don't use PHP and it's not obvious to me how to use your example to replicate the backend in other language. In general, it would be great to have an OpenAPI spec of the backend API (at least covering some of the most common operations) such that one could more easily put together a backend implementation. Perhaps this kind request belongs more to https://github.com/n1crack/vuefinder-php . Thanks for your attention.

**Reply by n1crack** [reply]
Hi,

Thank you for your kind comment! I plan to update the PHP code and create documentation for other languages. And I will definitely consider OpenAPI, thanks for the suggestion. My time is currently limited, but I aim to complete this as soon as possible. Thank you for your suggestion and understanding.

Best regards.

**Comment by n1crack** [comment]
https://vuefinder.ozdemir.be/api-reference/openapi.html
